neatnet/nodes.py

```diff
@@ -466,9 +467,35 @@ def consolidate_nodes(
             return gdf
 
     # get clusters of nodes which should be consolidated
-    linkage = hierarchy.linkage(nodes.get_coordinates(), method="average")
-    nodes["lab"] = hierarchy.fcluster(linkage, tolerance, criterion="distance")
-    unique, counts = np.unique(nodes["lab"], return_counts=True)
+    indices = nodes.sindex.query(
+        nodes.geometry, predicate="dwithin", distance=tolerance
+    )
+    matrix = sparse.coo_array(
+        (np.ones(len(indices[0]), dtype=np.bool_), indices[::-1]),
+        shape=(len(nodes), len(nodes)),
+        dtype=np.bool_,
+    )
+    dwithin_graph = graph.Graph.from_sparse(matrix, ids=nodes.index)
+    comp_labels = dwithin_graph.component_labels
+    cardinalities = dwithin_graph.cardinalities
+
+    mask = cardinalities > 1
+    components = comp_labels[mask]
+    nodes_to_merge = nodes[mask]
+
+    def get_labels(nodes):
+        linkage = hierarchy.linkage(shapely.get_coordinates(nodes), method="average")
+        labels = (
+            hierarchy.fcluster(linkage, tolerance, criterion="distance").astype(str)
+            + f"_{nodes.name}"
```

Comment:
It seems that we need numpy 2 for this line

Reply:
And `numpy>2` is already pinned as per your earlier push.